solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -67,12 +73,33 @@ public CoordinatorHttpSolrCall(
     this.factory = factory;
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
+  protected void writeResponse(
+      SolrQueryResponse solrRsp, QueryResponseWriter responseWriter, Method reqMethod)
+      throws IOException {
+
+    // make coordinator mode explicit
+    if (solrRsp.getValues().get(CommonParams.DEBUG) != null) {
+      final NamedList<Object> debug =
+          (NamedList<Object>) solrRsp.getValues().get(CommonParams.DEBUG);
```

Comment:
Maybe have local variable to avoid double `get` calls e.g.
```suggestion
    final Object debugObj = solrRsp.getValues().get(CommonParams.DEBUG);
    if (debugObj != null) {
      final NamedList<Object> debug =
          (NamedList<Object>) debugObj;
```